chemprop/uncertainty/predictor.py

```diff
@@ -101,10 +117,80 @@ def __call__(
 
 @UncertaintyPredictorRegistry.register("dropout")
 class DropoutPredictor(UncertaintyPredictor):
+    """
+    A :class:`DropoutPredictor` creates a virtual ensemble of models via Monte Carlo dropout with
+    the provided model [1]_.
+
+    References
+    -----------
+    .. [1] arXiv:1506.02142 [stat.ML]
+
+    Parameters
+    ----------
+    ensemble_size: int
+        The number of samples to draw for the ensemble.
+    dropout: float | None
+        The probability of dropping out units in the dropout layers. If unspecified,
+        the training probability is used, which is prefered but not possible if the model was not
+        trained with dropout (i.e. p=0).
+    """
+
+    def __init__(self, ensemble_size: int, dropout: None | float = None):
+        self.ensemble_size = ensemble_size
+        self.dropout = dropout
+
     def __call__(
         self, dataloader: DataLoader, models: Iterable[MPNN], trainer: pl.Trainer
     ) -> tuple[Tensor, Tensor]:
-        return
+        if len(models) != 1:
+            raise ValueError("Dropout method for uncertainty only takes exactly one model.")
+        model = next(iter(models))
+        self._setup_model(model)
+        individual_preds = []
+
+        for _ in range(self.ensemble_size):
+            predss = trainer.predict(model, dataloader)
+            preds = torch.concat(predss, 0)
+            individual_preds.append(preds)
+
+        stacked_preds = torch.stack(individual_preds, dim=0).float()
+        means = torch.mean(stacked_preds, dim=0).unsqueeze(0)
```

Comment:
I'm not sure if we should provide the average predictions here or individual predictions. Following the docstring, we probably should use the average, as we are only using one model. On the other hand, I haven't seen papers explicitly using the individual predictions from MC dropout to do something else, so I think we're fine.

Reply:
mean and uncertainty seem like good enough coverage to me